avro-fastserde/src/main/java/com/linkedin/avro/fastserde/ByteBufferBackedPrimitiveFloatList.java

```diff
@@ -185,8 +185,11 @@ public float getPrimitive(int i) {
     if (i >= size) {
       throw new IndexOutOfBoundsException("Index " + i + " out of bounds.");
     }
-    cacheFromByteBuffer();
-    return elements[i];
+
+    if (isCached) {
+      return elements[i];
+    }
+    return elements[i] = byteBuffer.getElement(i);
```

Comment:
nit - assigning the result to elements[i] doesnt do anything and can be dropped

Reply:
+1. I guess @majisourav99 was C++ programmer ;)

Reply:
yes Bingfeng, you are correct :) I will make it more expressed.